tests/unit/test_charm.py

```diff
@@ -61,431 +115,287 @@ def _read_file(path: str) -> str:
     def test_given_certificate_is_not_stored_when_configure_sdcore_smf_then_status_is_waiting(  # noqa: E501
-        self,
-        patch_nrf_url,
-        patch_check_output,
+        self, add_storage, mock_default_values, nrf_relation_id, certificates_relation_id
     ):
-        self.harness.add_storage("config", attach=True)
-        self.harness.add_storage("certs", attach=True)
         self.harness.set_can_connect(container=self.container_name, val=True)
-        patch_nrf_url.return_value = VALID_NRF_URL
-        patch_check_output.return_value = POD_IP
         self._create_database_relation_and_populate_data()
-        self._create_nrf_relation()
-        self._create_certificates_relation()
         self.harness.charm._storage_is_attached = Mock(return_value=True)
         self.harness.charm._ue_config_file_is_written = Mock(return_value=True)
         self.harness.charm._certificate_is_stored = Mock(return_value=False)
 
         self.harness.charm._configure_sdcore_smf(event=Mock())
         self.harness.evaluate_status()
-        self.assertEqual(
-            self.harness.model.unit.status, WaitingStatus("Waiting for certificates to be stored")
-        )
+        assert self.harness.model.unit.status == WaitingStatus("Waiting for certificates to be stored")  # noqa: E501
 
-    @patch(f"{CERTIFICATES_LIB}.get_assigned_certificates")
-    @patch("charm.check_output")
-    @patch("charms.sdcore_nrf_k8s.v0.fiveg_nrf.NRFRequires.nrf_url", new_callable=PropertyMock)
-    @patch("ops.model.Container.restart")
     def test_given_config_files_and_relations_are_created_when_configure_sdcore_smf_is_called_then_status_is_active(  # noqa: E501
-        self, _, patched_nrf_url, patch_check_output, patch_get_assigned_certificates
+        self, add_storage, mock_default_values, nrf_relation_id, certificates_relation_id
     ):
-        self.harness.add_storage(storage_name="certs", attach=True)
-        self.harness.add_storage(storage_name="config", attach=True)
         root = self.harness.get_filesystem_root(self.container_name)
-        provider_certificate = Mock(ProviderCertificate)
-        provider_certificate.certificate = CERTIFICATE
-        provider_certificate.csr = CSR
-        patch_get_assigned_certificates.return_value = [provider_certificate]
-
-        (root / "support/TLS/smf.pem").write_text(CERTIFICATE)
-        (root / "support/TLS/smf.csr").write_text(CSR)
-        (root / "etc/smf/uerouting.yaml").write_text(self._read_file("src/uerouting.yaml"))
+        self.mock_get_assigned_certificates.return_value = [self._get_provider_certificate()]
+        (root / CERTIFICATE_PATH).write_text(CERTIFICATE)
+        (root / CSR_PATH).write_text(CSR)
+        (root / UE_CONFIG_FILE_PATH).write_text(self._read_file(EXPECTED_UE_CONFIG_FILE_PATH))
         pod_ip = "1.1.1.1"
-        patch_check_output.return_value = pod_ip.encode()
+        self.mock_check_output.return_value = pod_ip.encode()
```

Comment:
this is already included in `mock_default_values`

Reply:
Done